server/src/index.ts

```
@@ -11,66 +12,111 @@ import {
+      params.append('scope', 'identify');
 
-      const oauthData = tokenResponseData.data;
-
-      const userResult = await axios({
-        method: 'GET',
-        url: 'https://discord.com/api/users/@me',
+      tokenResponseData = (await axios({
+        method: 'POST',
+        url: 'https://discord.com/api/oauth2/token',
+        data: params,
         headers: {
-          authorization: `${oauthData.token_type} ${oauthData.access_token}`,
-        }
-      });
+          'Content-Type': 'application/x-www-form-urlencoded'
+        },
+      })).data;
+
+      try {
+
+        const oauthData: DiscordOAuthTokenResponseData = tokenResponseData;
+
+        const userResult: DiscordUserData = (await axios({
+          method: 'GET',
+          url: 'https://discord.com/api/users/@me',
+          headers: {
+            authorization: `${oauthData.token_type} ${oauthData.access_token}`,
+          }
+        })).data;
 
-      const { username, discriminator } = userResult.data;
-      console.log(`Hi ${username}${discriminator}`)
+        const { username, discriminator } = userResult;
 
-      const token = createToken(`${username}${discriminator}`);
-      return res.json({ token })
+        const token = createToken(`${username}${discriminator}`);
+        return res.json({ token });
+      } catch (error) {
+
+        error.response.config.headers.authorization = "Bearer REDACTED";
+        error.response.request._header = "REDACTED";
+
+        error.response.message = error.response.data.message;
+        next(error.response);
+      }
     } catch (error) {
-      console.error(error);
+      const { data } = error.response.config;
+      error.response.config.data =
+        `client_id=REDACTED&client_secret=REDACTED&${data.substring(data.indexOf("grant_type"))}`;
+
+      error.response.message = error.response.data.error_description;
+      next(error.response);
     }
+  } else {
+    return next(new BadRequestError());
```

should we move this up top so the `if (code) {}` condition can be unindented one level?

eg. 
```js
if (!code) {
  return next(new BadRequestError());
}

let tokenResponseData: DiscordOAuthTokenResponseData;
...
```

---

Ah fail fast, made the change!